paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RemoveOrphanFilesAction.java

```diff
@@ -18,51 +18,121 @@
 
     public RemoveOrphanFilesAction(
             String warehouse,
             String databaseName,
             String tableName,
-            Map<String, String> catalogConfig) {
-        super(warehouse, databaseName, tableName, catalogConfig);
+            Map<String, String> catalogConfig)
+            throws Catalog.TableNotExistException, Catalog.DatabaseNotExistException {
+        super(warehouse, catalogConfig);
+        this.tableOrphanFilesCleans = new ArrayList<>();
+        constructOrphanFilesCleans(catalog, databaseName, tableName, tableOrphanFilesCleans);
+    }
 
+    public static void constructOrphanFilesCleans(
+            Catalog catalog,
+            String databaseName,
+            String tableName,
+            List<Pair<String, OrphanFilesClean>> tableOrphanFilesCleans)
+            throws Catalog.DatabaseNotExistException, Catalog.TableNotExistException {
+        if ("*".equals(tableName)) {
+            for (String t : catalog.listTables(databaseName)) {
+                constructOrphanFilesCleanInternal(catalog, databaseName, t, tableOrphanFilesCleans);
+            }
+        } else {
+            constructOrphanFilesCleanInternal(
+                    catalog, databaseName, tableName, tableOrphanFilesCleans);
+        }
+    }
+
+    public static void constructOrphanFilesCleanInternal(
+            Catalog catalog,
+            String databaseName,
+            String tableName,
+            List<Pair<String, OrphanFilesClean>> tableOrphanFilesCleans)
+            throws Catalog.TableNotExistException {
+        Identifier identifier = new Identifier(databaseName, tableName);
+        Table table = catalog.getTable(identifier);
         checkArgument(
                 table instanceof FileStoreTable,
                 "Only FileStoreTable supports remove-orphan-files action. The table type is '%s'.",
                 table.getClass().getName());
-        this.orphanFilesClean = new OrphanFilesClean((FileStoreTable) table);
+
+        tableOrphanFilesCleans.add(
+                Pair.of(tableName, new OrphanFilesClean((FileStoreTable) table)));
+    }
+
+    public void olderThan(String olderThan) {
+        this.tableOrphanFilesCleans.forEach(
+                orphanFilesClean -> orphanFilesClean.getRight().olderThan(olderThan));
     }
 
-    public RemoveOrphanFilesAction olderThan(String timestamp) {
-        this.orphanFilesClean.olderThan(timestamp);
-        return this;
+    public void dryRun() {
+        this.tableOrphanFilesCleans.forEach(
+                orphanFilesClean -> orphanFilesClean.getRight().fileCleaner(path -> {}));
     }
 
-    public RemoveOrphanFilesAction dryRun() {
-        this.orphanFilesClean.fileCleaner(path -> {});
-        return this;
+    public static String[] doCleanOrphanFiles(
+            List<Pair<String, OrphanFilesClean>> tableOrphanFilesCleans) {
+        ForkJoinPool executePool = ScanParallelExecutor.getExecutePool(null);
```

Comment:
Just create a new pool to avoid dead lock.

Reply:
done